obspy/core/tests/test_trace.py

```diff
@@ -2203,6 +2203,269 @@ def test_remove_response_plot(self):
+            np.int32)
+        self.assertEqual(
+            tr_int64.copy().decimate(factor=2, no_filter=True).data.dtype,
+            np.int64)
+        self.assertEqual(
+            tr_float32.copy().decimate(factor=2, no_filter=True).data.dtype,
+            np.float32)
+        self.assertEqual(
+            tr_float64.copy().decimate(factor=2, no_filter=True).data.dtype,
+            np.float64)
+
+        # Detrending will upcast integers but should not touch floats.
+        self.assertEqual(tr_int32.copy().detrend("simple").data.dtype,
+                         np.float64)
+        self.assertEqual(tr_int64.copy().detrend("simple").data.dtype,
+                         np.float64)
+        self.assertEqual(tr_float32.copy().detrend("simple").data.dtype,
+                         np.float32)
+        self.assertEqual(tr_float64.copy().detrend("simple").data.dtype,
+                         np.float64)
+        self.assertEqual(tr_int32.copy().detrend("linear").data.dtype,
+                         np.float64)
+        self.assertEqual(tr_int64.copy().detrend("linear").data.dtype,
+                         np.float64)
+        self.assertEqual(tr_float32.copy().detrend("linear").data.dtype,
+                         np.float32)
+        self.assertEqual(tr_float64.copy().detrend("linear").data.dtype,
+                         np.float64)
+        self.assertEqual(tr_int32.copy().detrend("constant").data.dtype,
+                         np.float64)
+        self.assertEqual(tr_int64.copy().detrend("constant").data.dtype,
+                         np.float64)
+        self.assertEqual(tr_float32.copy().detrend("constant").data.dtype,
+                         np.float32)
+        self.assertEqual(tr_float64.copy().detrend("constant").data.dtype,
+                         np.float64)
+        self.assertEqual(
+            tr_int32.copy().detrend("polynomial", order=3).data.dtype,
+            np.float64)
+        self.assertEqual(
+            tr_int64.copy().detrend("polynomial", order=3).data.dtype,
+            np.float64)
+        self.assertEqual(
+            tr_float32.copy().detrend("polynomial", order=3).data.dtype,
+            np.float32)
+        self.assertEqual(
+            tr_float64.copy().detrend("polynomial", order=3).data.dtype,
+            np.float64)
+        self.assertEqual(
+            tr_int32.copy().detrend("spline", order=3, dspline=100).data.dtype,
+            np.float64)
+        self.assertEqual(
+            tr_int64.copy().detrend("spline", order=3, dspline=100).data.dtype,
+            np.float64)
+        self.assertEqual(
+            tr_float32.copy().detrend("spline", order=3,
+                                      dspline=100).data.dtype,
+            np.float32)
+        self.assertEqual(
+            tr_float64.copy().detrend("spline", order=3,
+                                      dspline=100).data.dtype,
+            np.float64)
+
+        # Tapering.
```

Comment:
These four don't have an explanation of what they should do (though it's kind of obvious).